src/riak_kv_leveled_backend.erl

```diff
@@ -400,31 +408,41 @@ fold_objects(FoldObjectsFun, Acc, Opts, #state{bookie=Bookie}) ->
                         false ->
                             false
                     end,
+                
                 SpecialFoldFun = 
                     fun(ObjB, ObjK, Obj, InnerAcc) ->
                         case riak_index:object_key_in_range({ObjB, ObjK}, 
                                                             FilterBucket, Q) of
                             {true, _BK} ->
                                 case StndObjFold of   
                                     true ->
-                                        FoldObjectsFun(ObjB, ObjK, Obj, 
+                                        FoldObjectsFun(ObjB,
+                                                        ObjK,
+                                                        Obj, 
                                                         InnerAcc);
                                     false ->
                                         % Assumption here is that if this is 
                                         % not flagged as a standard object fold
                                         % it is using a fold_keys_fun -
                                         % so the object is disguised as a key
-                                        FoldObjectsFun(ObjB, {o, ObjK, Obj}, 
+                                        FoldObjectsFun(ObjB,
+                                                        {o, ObjK, Obj}, 
                                                         InnerAcc)
                                 end;
                             {skip, _BK} ->
-                                Acc
+                                InnerAcc;
+                            _ ->
+                                % Required for riak_test/tests/verify_cs_bucket
+                                throw(InnerAcc)
                         end
                     end,
+                % StartKey and StartInclusive based on query, but the EndKey
+                % and EndInclusive should be handled by the passed in fold
+                % function, so null is used for EndKey
                 leveled_bookie:book_objectfold(Bookie, 
                                                 ?RIAK_TAG,
                                                 FilterBucket, 
-                                                all, 
+                                                {StartKey, null}, 
```

Comment:
The commend talks about EndKey, but here StartKey is `null` that is confusing

Reply:
I guess this is an interval from StartKey to`null`

Reply:
Yes, null means there's no end to the range - so it will stop on the first object out of range (or the end of the bucket).

There is perhaps a more general problem here about end_inclusive.  riak_kv_eleveldb_backend uses the riak_index range checkers and a throw for most queries, so that end_inclusive and start_inclusive is controlled via the those range checkers (and at the end of the range this means handling the state false).

In other queries the leveled backend hasn't used these checks, and relied on an end key in the range.  However, this then misses the control about end_inclusive.  It is only in this CS fold we have discovered the end_inclusive being used, so it is only here we rely on the range checker not setting an end_key to finish the fold.

In other folds, leveled backends will ignore end_inclusive, but leveldb backends will respect end_inclusive - but we don't know if that is actually a problem (if other folds don't set end_inclusive).

I will raise an issue to investigate and potentially correct this anomaly, but for now, as this is intended to patch a release I'm focusing on the minimal change to correct the known usage of end_inclusive - rather than making. amore general change to use riak_index range checkers.